GitHubBranches.ps1

```diff
@@ -154,6 +154,518 @@ filter Get-GitHubRepositoryBranch
+        [int] $RequiredApprovingReviewCount,
+
+        [string[]] $RestrictPushUsers,
+
+        [string[]] $RestrictPushTeams,
+
+        [string[]] $RestrictPushApps,
+
+        [switch]$RequiredLinearHistory,
+
+        [switch] $AllowForcePushes,
+
+        [switch] $AllowDeletions,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        OwnerName = (Get-PiiSafeString -PlainText $OwnerName)
+        RepositoryName = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    if ($PSBoundParameters.ContainsKey('StatusChecks'))
+    {
+        $requiredStatusChecks = @{
+            strict = $RequireUpToDateBranches.ToBool()
+            contexts = $StatusChecks
+        }
+    }
+    else
+    {
+        $requiredStatusChecks = $null
+    }
+
+    $dismissalRestrictions = @{}
+
+    if ($PSBoundParameters.ContainsKey('DismissalUsers')) {
+       $dismissalRestrictions['users'] = $DismissalUsers }
+    if ($PSBoundParameters.ContainsKey('DismissalTeams'))
+    {
+        $teams = Get-GitHubTeam -OwnerName $OwnerName -RepositoryName $RepositoryName |
+            Where-Object -FilterScript { $DismissalTeams -contains $_.name }
+        $dismissalRestrictions['teams'] = @($teams.slug)
+    }
+
+    $requiredPullRequestReviews = @{}
+
+    if ($PSBoundParameters.ContainsKey('DismissStaleReviews')) {
+        $requiredPullRequestReviews['dismiss_stale_reviews'] = $DismissStaleReviews.ToBool() }
+    if ($PSBoundParameters.ContainsKey('RequireCodeOwnerReviews')) {
+        $requiredPullRequestReviews['require_code_owner_reviews'] = $RequireCodeOwnerReviews.ToBool() }
+    if ($PSBoundParameters.ContainsKey('RequiredApprovingReviewCount')) {
+        $requiredPullRequestReviews['required_approving_review_count'] = $RequiredApprovingReviewCount }
+
+    if ($dismissalRestrictions.count -gt 0)
+    {
+        $requiredPullRequestReviews['dismissal_restrictions'] = $dismissalRestrictions
+    }
+
+    if ($requiredPullRequestReviews.count -eq 0)
+    {
+        $requiredPullRequestReviews = $null
+    }
+
+    if ($PSBoundParameters.ContainsKey('RestrictPushUsers') -or
+        $PSBoundParameters.ContainsKey('RestrictPushTeams') -or
+        $PSBoundParameters.ContainsKey('RestrictPushApps'))
+    {
+        if ($null -eq $RestrictPushUsers)
+        {
+            $RestrictPushUsers = @()
+        }
+
+        if ($null -eq $RestrictPushTeams)
+        {
+            $RestrictPushTeams = @()
+        }
+
+        $restrictions = @{
+            users = $RestrictPushUsers
+            teams = $RestrictPushTeams
+        }
+
+        if ($PSBoundParameters.ContainsKey('RestrictPushApps')) {
+            $restrictions['apps'] = $RestrictPushApps }
+    }
+    else
+    {
+        $restrictions = $null
+    }
+
+    $hashBody = @{
+        name = $RepositoryName
```

Comment:
I don't see this being part of the spec for the request body

Reply:
Removed